### PR TITLE
Display credentials

### PR DESCRIPTION
**Pull request by orhoj**
## Purpose
To display credentials.

## Changes
- New page for showing a list of credentials.

## Checklist
- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: packages/browser-wallet/src/popup/pages/VerifiableCredential/VerifiableCredentialCard.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        type: 'CredentialSchema2022', // the same for all schemas
+    },
+};
+
+export const Primary: ComponentStory<typeof VerifiableCredentialCard> = () => {
+    return (
+        <div style={{ width: 375 }}>
+            <VerifiableCredentialCard
+                credential={verifiableCredential}
+                schema={schema}
```
Comment (shjortConcordium) [inline packages/browser-wallet/src/popup/pages/VerifiableCredential/VerifiableCredentialCard.stories.tsx:92]: Does this compile? :thinking:

File: packages/browser-wallet/src/shared/storage/types.ts [hunk 2]
```diff
@@ -254,3 +257,53 @@ export type AcceptedTermsState = {
...
+    Expired,
+    NotActivated,
+}
+
+interface CredentialSchema {
+    id: string;
+    type: string;
+}
+
+export type CredentialSubject = { id: string } & Record<string, string | number>;
```
[the remark this answers is not in the document]
Comment (orhoj) [reply packages/browser-wallet/src/shared/storage/types.ts:273]: Okay, I will leave it as is as you have fixed it already.

File: packages/browser-wallet/src/shared/utils/verifiable-credential-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,633 @@
...
+import jsonschema from 'jsonschema';
+import { getContractName } from './contract-helpers';
+
+/**
+ * Extracts the credential holder id from a verifiable credential id (did).
+ * @param credentialId the did for a credential
+ * @returns the credential holder id
+ */
+export function getCredentialHolderId(credentialId: string): string {
+    const splitted = credentialId.split('/');
```
[the remark this answers is not in the document]
Comment (orhoj) [reply packages/browser-wallet/src/shared/utils/verifiable-credential-helpers.ts:18]: "(nonstandard or archaic) Simple past tense and past participle of split."

But I agree that it is not very good wording. I have changed it to `credentialIdSplit`.

File: packages/browser-wallet/src/popup/pages/VerifiableCredential/VerifiableCredentialHooks.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+    useEffect(() => {
+        if (!schemas.loading) {
+            const schemaValue = schemas.value[credential.credentialSchema.id];
+            if (!schemaValue) {
+                throw new Error(`Attempted to find schema for credentialId: ${credential.id} but none was found!`);
+            }
+            setSchema(schemaValue);
+        }
+    }, [schemas]);
```
[the remark this answers is not in the document]
Comment (orhoj) [reply packages/browser-wallet/src/popup/pages/VerifiableCredential/VerifiableCredentialHooks.tsx:57]: Will depend on `schemas.loading`.

File: packages/browser-wallet/src/shared/utils/verifiable-credential-helpers.ts [hunk 5]
```diff
@@ -0,0 +1,633 @@
...
+ */
+export async function getCredentialSchemas(
+    credentials: VerifiableCredential[],
+    abortControllers: AbortController[],
+    client: ConcordiumGRPCClient
+) {
+    const onChainSchemas: VerifiableCredentialSchema[] = [];
+
+    const allContractAddresses = credentials.map((vc) => getCredentialRegistryContractAddress(vc.id));
+    const issuerContractAddresses = [...new Set(allContractAddresses)];
```
[the remark this answers is not in the document]
Comment (orhoj) [reply packages/browser-wallet/src/shared/utils/verifiable-credential-helpers.ts:505]: Yes, I can!

File: packages/browser-wallet/src/popup/pages/VerifiableCredential/VerifiableCredentialCard.stories.tsx (hunk 1 above)
Comment (orhoj) [reply packages/browser-wallet/src/popup/pages/VerifiableCredential/VerifiableCredentialCard.stories.tsx:92]: Yes.

File: packages/browser-wallet/src/shared/utils/verifiable-credential-helpers.ts [hunk 6]
```diff
@@ -0,0 +1,633 @@
...
+    credentials: VerifiableCredential[],
+    client: ConcordiumGRPCClient,
+    abortControllers: AbortController[],
+    storedSchemas: Record<string, VerifiableCredentialSchema>
+) {
+    const upToDateSchemas = await getCredentialSchemas(credentials, abortControllers, client);
+    let updatedSchemasInStorage = { ...storedSchemas };
+    let updateReceived = false;
+
+    // TODO Verify that something has actually changed before making the update.
```
[the remark this answers is not in the document]
Comment (orhoj) [reply packages/browser-wallet/src/shared/utils/verifiable-credential-helpers.ts:618]: The comment should have been deleted.

**Review by shjortConcordium — APPROVED** [review]

**Pull request merged**